File: 111. Minimum Depth of Binary Tree/minimumDepthOfBinaryTree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
from collections import deque


class Solution:
    def minDepth(self, root: TreeNode) -> int:
        if root == None:
            return 0
        q = deque()
        q.appendleft(root)
        result = 0
        found = False
        while q:
            l = len(q)
            result += 1
            for _ in range(l):
                node = q.pop()
                if node.left == None and node.right == None:
                    return result
                if node.left != None:
                    q.appendleft(node.left)
                if node.right != None:
                    q.appendleft(node.right)
        return result
                
        
        

File: 111. Minimum Depth of Binary Tree/minimumDepthOfBinaryTree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
#include <queue>

class Solution {
public:
    int minDepth(TreeNode* root) {
        if(root == NULL) return 0;
        int result = 0;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int size = q.size();
            result++;
            for(int i = 0; i < size; i++){
                auto node = q.front();
                q.pop();
                if(node->left == NULL && node->right == NULL) return result;
                if(node->left != NULL) q.push(node->left);
                if(node->right != NULL) q.push(node->right);
            }
        }
        return result;
        
    }
};
